src/store/proposals/proposals.sagas.ts

```diff
@@ -108,10 +115,69 @@ const watchProposalsList = createWatcherSaga({
   stopAction: proposalsActions.stopWatchingProposalsList.type,
 });
 
+export function* addProposal({ payload }: PayloadAction<AddProposalFields>) {
+  try {
+    const account = yield* select(accountSelector);
+    const staking = yield* select(stakingContractSelector);
+    const isGovAdmin =
+      payload[AddProposalInputs.SendProposalContract] ===
+      GOVERNANCE_OPTIONS.GOVERNER_ADMIN.id;
+
+    const govSelector = isGovAdmin
+      ? governorAdminSelector
+      : governorOwnerSelector;
+
+    const governor = yield* select(govSelector);
+
+    if (!account || !governor || !staking) {
+      throw new Error('Wallet not connected');
+    }
+
+    const threshold = yield* call(governor.proposalThreshold);
+    const votes = yield* call(staking.getCurrentVotes, account);
+
+    if (threshold.gt(votes)) {
+      const errorMsg = `Your voting power must be at least ${threshold.toString()} to make a proposal`;
+      yield* put(proposalsActions.porposalFailure(errorMsg));
+      return;
+    }
+
+    const rows = payload[AddProposalInputs.Values];
+
+    const targets = rows.map((row) => row[AddProposalInputs.Target]);
+    const values: BigNumberish[] = rows.map(
+      (row) => row[AddProposalInputs.Value]
+    );
+    const signatures = rows.map((row) => row[AddProposalInputs.Signature]);
+    const calldatas: BytesLike[] = rows.map(
+      (row) => row[AddProposalInputs.Signature]
+    );
+    const description = payload[AddProposalInputs.Description];
+
+    yield* call(
```

Comment:
Please wait for the transaction to be put in a block before dispatching a success action. It would look something like that:
```suggestion
        const tx = yield* call(
      governor.propose,
      targets,
      values,
      signatures,
      calldatas,
      description
    );

    yield* call(tx.wait);
```

Reply:
`transaction.wait`

https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse